packages/main/src/Popover.ts

```diff
@@ -492,6 +503,20 @@ class Popover extends Popup {
 		return top + (Number.parseInt(this.style.top || "0") - actualTop);
 	}
 
+	_getContainingBlockClientRect() {
+		let parentNode = this.parentElement ? this.parentNode as HTMLElement : (this.parentNode as ShadowRoot).host as HTMLElement;
+
+		while (parentNode) {
+			if (isElementContainingBlock(parentNode)) {
+				return parentNode.getBoundingClientRect();
+			}
+
+			parentNode = parentNode.parentElement ? parentNode.parentNode as HTMLElement : (parentNode.parentNode as ShadowRoot).host as HTMLElement;
+		}
+
+		return { left: 0, top: 0 };
```

Comment:
Maybe returning null here could be better. Or otherwise including all the fields that a getBoundingClientRect() returns can make it more consistent with the name of the method.
When an element is not placed in the document, for example `document.createElement("a").getBoundingClientRect()`, the result is all zeroes: `{
    "x": 0,
    "y": 0,
    "width": 0,
    "height": 0,
    "top": 0,
    "right": 0,
    "bottom": 0,
    "left": 0
}`.